_security/access-control/resources.md

```diff
@@ -0,0 +1,173 @@
+```
+{% include copy.html %}
+
+### API
+
+All sharing metadata is stored in the `.opensearch_resource_sharing` system index, as shown in the following table.
+
+| Field | Type | Description |
+| :--- | :--- | :--- |
+| `source_idx`  | String | The system index holding the resource |
+| `resource_id` | String | The resource ID   |
+| `created_by`  | Object | The name of the user who created the resource    |
+| `share_with`  | Object | A map of `action-groups` to access definitions  |
+
+#### Java client APIs
+
+Use the following classes when interacting with the extension through the Java client:
+
+```java
+ResourceSharingClient client = ResourceSharingClientAccessor.getResourceSharingClient();
+
+// Verify access
+client.verifyResourceAccess(resourceId, indexName, listener);
+
+// Share resource
+client.share(resourceId, indexName, recipients, listener);
+
+// Revoke access
+client.revoke(resourceId, indexName, recipients, listener);
+
+// List accessible IDs
+client.getAccessibleResourceIds(indexName, listener);
+```
+{% include copy.html %}
+
+### Document structure
+
+The resource sharing metadata is stored as JSON documents that define ownership, permissions, and access patterns. Each document contains fields that specify the resource location, identifier, creator information, and sharing configuration:
+
+```json
+{
+  "source_idx": ".plugins-ml-model-group",
+  "resource_id": "model-group-123",
+  "created_by": {
+    "user": "darshit"
+  },
+  "share_with": {
+    "default": {
+      "users": ["user1"],
+      "roles": ["viewer_role"],
+      "backend_roles": ["data_analyst"]
+    }
+  }
+}
+```
+{% include copy.html %}
+
+## Action groups
+
+Action groups define permission levels for shared resources. As of OpenSearch 3.0, the only supposted action group for resource sharing is `default`. To share resources across the action groups, use the `share_with` array in that resource's configuration and add wildcards for each default role:
+
+```json
+{
+  "share_with": {
+    "default": {
+      "users": ["*"],
+      "roles": ["*"],
+      "backend_roles": ["*"]
+    }
+  }
+}
+```
+{% include copy.html %}
+
+To keep a resource private, keep the `share_with` array empty:
+
+```json
+{ "share_with": {} }
+```
+{% include copy.html %}
+
+## Restrictions
+
+Before implementing resource sharing and access control, be aware of the following limitations that help maintain security and consistency across the system:
+
+- Only resource owners or super-admins can share or revoke access.
+- Resources must reside in system indexes.
+- Disabling system index protection exposes resources to direct index-level access.
+
+## Best practices
+
+To ensure secure and efficient implementation of resource sharing and access control, follow these recommended practices:
+
+- Declare and register extensions correctly.
+- Use the SPI client APIs instead of manual index queries.
+- Enable only on fresh 3.0.0+ clusters to avoid upgrade issues.
```

Comment:
What do we mean by "fresh" clusters?

Reply:
This feature is intended to be enabled only on brand new clusters.

Reply:
In next iteration, we will add a migration path to enable usage on existing clusters.